alfresco-solr-query-analytics-6x/src/main/java/eu/xenit/alfresco/instrumentation/solr/DebugSolrQueryHTTPClientAdvisor.java

```diff
@@ -85,8 +87,12 @@ public Object invoke(MethodInvocation mi) throws Throwable {
                             }
                         } catch (Throwable e) {
                             String queryString = params.getQuery();
+                            StringWriter sw = new StringWriter();
+                            PrintWriter pw = new PrintWriter(sw);
+                            e.printStackTrace(pw);
+                            String sStackTrace = sw.toString(); // stack trace as a string
                             logger.debug(
-                                    "{\"parsedQuery\":" + queryString + ", \"debugError\":" + e.getMessage() + "}");
+                                    "{\"parsedQuery\":" + queryString + ", \"debugError\":" + DebugSolrQueryHTTPClientAdvisor.this.solrQueryParser.escapeIllegalChars(sStackTrace) + "}");
```

Comment:
Why is this reference written like this?
Why not this.solrQueryParser.escapeIllegalChars(sStackTrace);
or solrQueryParser.escapeIllegalChars(sStackTrace)

(I'm not used to this notation, it looks like Static reference ?

Reply:
I just copied it over from the first reference to the class and verified that it works. I don't really know why the notation is like that maybe some bean nameclash prevention.